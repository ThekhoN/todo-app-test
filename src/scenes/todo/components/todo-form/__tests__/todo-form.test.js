import React from "react";
import { shallow, mount } from "enzyme";
import TodoForm from "../index";

describe("TodoForm", () => {
	let onSubmit, wrapper, wrapperMounted;
	beforeEach(() => {
		onSubmit = jest.fn();
		wrapper = shallow(<TodoForm onSubmit={onSubmit} />);
		wrapperMounted = mount(<TodoForm onSubmit={onSubmit} />);
	});

	it("matches snapshot", () => {
		expect(wrapper).toMatchSnapshot();
	});

	it("runs onSubmit on form submit", () => {
		wrapper.simulate("submit");
		expect(onSubmit).toBeCalled();
	});

	it("runs onSubmit on click submit button", () => {
		const todoSubmitButton = wrapperMounted.find("button");
		todoSubmitButton.simulate("click");
		// onSubmit is props.onSubmit
		expect(onSubmit).toBeCalled();
	});

	it("runs onInputChange on input change", () => {
		// onInputChange is an own method -> this.onInputChange
		wrapperMounted.instance().onInputChange = jest.fn();
		wrapperMounted.instance().forceUpdate();
		wrapperMounted.find("input").simulate("change");
		expect(wrapperMounted.instance().onInputChange).toBeCalled();
	});
});
