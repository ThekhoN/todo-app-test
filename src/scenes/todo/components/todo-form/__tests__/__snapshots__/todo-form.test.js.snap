// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm
    onSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TodoInput
          onChange={[Function]}
        />,
        <TodoSubmitButton
          labelText="Add Todo"
          onClick={[Function]}
        />,
      ],
      "className": "todo-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "labelText": "Add Todo",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TodoInput
            onChange={[Function]}
          />,
          <TodoSubmitButton
            labelText="Add Todo"
            onClick={[Function]}
          />,
        ],
        "className": "todo-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "labelText": "Add Todo",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
